python/cudf/cudf/tests/test_parquet.py

```diff
@@ -3040,3 +3040,14 @@ def test_parquet_reader_multiindex():
 def test_parquet_reader_engine_error():
     with pytest.raises(ValueError):
         cudf.read_parquet(BytesIO(), engine="abc")
+
+
+def test_parquet_reader_zstd_huff_tables(datadir):
+    # Ensure that this zstd-compressed file does not overrun buffers. The
+    # problem was fixed in nvcomp 3.0.6.
+    # See https://github.com/rapidsai/cudf/issues/15096
+    fname = datadir / "zstd_huff_tables_bug.parquet"
+
+    # expect a failure when reading the whole file
+    with pytest.raises(RuntimeError):
+        cudf.read_parquet(fname)
```

Comment:
TODO: This needs to be modified (I just don't have time at this moment). We probably want to read the file with another reader implementation and compare the results?

Reply:
I verified that `compute-sanitizer` fails with `nvcomp 3.0.5` and passes with `nvcomp 3.0.6`. This is indicative that the fix is good.